### DISCUSSION TITLE
`preload` for `GET` forms

### DISCUSSION THREAD

**Discussion by chrisdone** [bigskysoftware/htmx#1465] [Ideas]
**Example**

Here's a simple "put in some code and see the reply below" form. The evaluator is pure (for a subset of Haskell), and therefore a POST/PUT is not correct, a GET is, and the result should be cached (because it will never change). That's all good. I can achieve that with `hx-include`, as seen below.

```html
<form hx-include="find [name='code']" hx-get="/evaluator" hx-target="find .reply">
  <textarea name="code"></textarea>
  <div class="reply"></div>
  <button class="run-button">Run</button>
</form>
```

**Current behavior**

As it stands, I can only specify `preload="mousedown"`, but two things happen:

1. This applies to the whole form, not to the submit button.
2. The `hx-include` is ignored (presumably because nobody expected I'd be trying to preload a form!)

**Desired behavior**

It's not super important to me that when a user clicks "Run" that it immediately shows a result, it just adds to an overall feeling of snappiness. But, it seems worth discussing.

Evaluation is cheap and quick, but the latency hit can only be mitigated by preloading.

What would be nice is to specify, e.g.

```html
<form .. preload="mousedown:find .run-button">
```

What do you think?

**Related links**

As it happens, there is an RFC to add [a `QUERY` method](https://httpwg.org/http-extensions/draft-ietf-httpbis-safe-method-w-body.html) which would effectively be a `GET` that lets you submit a body to it, which would be a lovely use-case for this kind of end-point. However, it's only an RFC.

**Comment by chrisdone** [comment]
Another way of writing it:

```html
<form>
  <textarea name="code" class="code"></textarea>
  <div class="reply"></div>
  <button hx-include="previous .code" hx-get="/evaluator" hx-target="previous .reply" class="">Run</button>
</form>
```

This fixes the `mousedown` issue. It's a little less clean than the 'find' (IMHO), but is okay on locality of behavior. **However, the `hx-include` issue remains.**

**Comment by chrisdone-artificial** [comment]
I think below is where we could pay attention to the `hx-include` property:

https://github.com/bigskysoftware/htmx/blob/6180ad2b25c9b40419ab1d10b124ba67c33c21a8/src/ext/preload.js#L49-L55
